src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -0,0 +1,149 @@
+    Preconditions.checkArgument(config.getServiceDirectories().size() > 0);
+    AppEngineDescriptor appengineWeb;
+    try {
+      // TODO(ludo: Make sure we support the case when more than 1 service is given...
+      FileInputStream is = new FileInputStream(
+              new File(config.getServiceDirectories().get(0), "WEB-INF/appengine-web.xml"));
+      appengineWeb = AppEngineDescriptor.parse(is);
+    } catch (IOException e) {
+      throw new AppEngineException(e);
+    }
+    List<String> arguments = new ArrayList<>();
+
+    Map<String, String> env = Maps.newHashMap();
+    if (!Strings.isNullOrEmpty(config.getJavaHomeDir())) {
+      env.put("JAVA_HOME", config.getJavaHomeDir());
+    }
+
+    List<String> jvmArguments = new ArrayList<>();
+    boolean isJava8 = (appengineWeb.getRuntime() != null)
+            && appengineWeb.getRuntime().startsWith("java8");
+    if (isJava8) {
+      jvmArguments.add("-Duse_jetty9_runtime=true");
+      jvmArguments.add("-D--enable_all_permissions=true");
+    } else {
+      // Add in the appengine agent
+      String appengineAgentJar = new File(
+              sdk.getJavaAppEngineSdkPath().toFile(), "agent/appengine-agent.jar")
+              .getAbsolutePath();
+      jvmArguments.add("-javaagent:" + appengineAgentJar);
+    } 
+    arguments.addAll(DevAppServerArgs.get("server", config.getHost()));
+    arguments.addAll(DevAppServerArgs.get("address", config.getPort()));
+    arguments.addAll(DevAppServerArgs.get("jvm_flag", config.getJvmFlags()));
+
+    arguments.add("--allow_remote_shutdown");
+    if (isJava8) {
+      arguments.add("--no_java_agent");
+    }
+    for (File service : config.getServiceDirectories()) {
+      arguments.add(service.toPath().toString());
+    }
+    try {
+      sdk.runDevAppServer1Command(jvmArguments, arguments, env);
+    } catch (ProcessRunnerException e) {
+      throw new AppEngineException(e);
+    }
+  }
+
+  /**
+   * Stops the local development server.
+   */
+  @Override
+  public void stop(StopConfiguration configuration) throws AppEngineException {
+    Preconditions.checkNotNull(configuration);
+
+    try {
+      URL adminServerUrl = new URL(
+              "http",
+              configuration.getAdminHost() != null
+              ? configuration.getAdminHost() : DEFAULT_HOST,
+              /* configuration.getAdminPort() != null ? configuration.getAdminPort() : */
+              DEFAULT_PORT,
+              "/_ah/admin/quit");
+      HttpURLConnection connection = (HttpURLConnection) adminServerUrl.openConnection();
+
+      connection.setDoOutput(true);
+      connection.setDoInput(true);
+      connection.setReadTimeout(4000);
+
+      connection.setRequestMethod("POST");
+      connection.getOutputStream().write(10);
+      connection.getInputStream();
+      connection.getOutputStream().flush();
+      connection.getOutputStream().close();
+      connection.getInputStream().close();
+
+      connection.connect();
```

Comment:
Strange. You're getting the inout stream before connecting. I think getInputStream implies a connection.